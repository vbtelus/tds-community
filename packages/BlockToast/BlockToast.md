(describe what your component does here)

### Usage criteria

- What must your experience achieve in order to warrant the use of this component?
- When using this component, how should you typically use it?

```jsx
<BlockToast
  headline="This is headline piece"
  copy="This is bloacktoast component"
  variant="default"
  link={{
    href: 'wwww.google.com',
    text: 'Click me',
  }}
/>
```
