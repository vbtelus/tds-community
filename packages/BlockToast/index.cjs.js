const BlockToast = require('./dist/index.cjs')

module.exports = BlockToast
