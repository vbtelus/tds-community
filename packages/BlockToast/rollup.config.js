import configure from '../../config/rollup.config'
import { dependencies } from './package.json'

export default configure({
  input: './BlockToast.jsx',
  dependencies,
})
