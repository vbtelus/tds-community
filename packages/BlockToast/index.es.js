import BlockToast from './dist/index.es'

export default BlockToast
