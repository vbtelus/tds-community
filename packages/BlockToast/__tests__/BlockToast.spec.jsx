import React from 'react'
import { shallow } from 'enzyme'

import BlockToast from '../BlockToast'

const STYLE_VARIANTS = {
  DEFAULT: 'default',
  TOAST: 'toast',
  LIGHT: 'light',
}
const headline = 'Limited time offer.'
const copy = 'Save $100 when you shop online'
const link = {
  href: 'https://www.telus.com',
  text: 'Learn now',
}

const { TOAST, LIGHT } = STYLE_VARIANTS

describe('BlockToast', () => {
  const doShallow = (props = {}) => shallow(<BlockToast copy={copy} {...props} />)

  it('renders', () => {
    const blockToast = doShallow()

    expect(blockToast).toMatchSnapshot()
  })

  it('does not allow custom CSS', () => {
    const blockToast = doShallow({
      className: 'my-custom-class',
      style: { color: 'hotpink' },
    })

    expect(blockToast).not.toHaveProp('className', 'my-custom-class')
    expect(blockToast).not.toHaveProp('style')
  })

  it('should render properly with both copy and link provided', () => {
    const blockToast = doShallow(link)
    expect(blockToast).toMatchSnapshot()
  })

  it('should render properly with headline, copy and link provided', () => {
    const updatedProp = {
      headline,
      link,
    }
    const blockToast = doShallow(updatedProp)
    expect(blockToast).toMatchSnapshot()
  })

  it('should render properly with toast variant', () => {
    const updatedProp = {
      headline,
      link,
      variant: TOAST,
    }
    const blockToast = doShallow(updatedProp)
    expect(blockToast).toMatchSnapshot()
  })

  it('should render properly with light variant', () => {
    const updatedProp = {
      headline,
      link,
      variant: LIGHT,
    }
    const blockToast = doShallow(updatedProp)
    expect(blockToast).toMatchSnapshot()
  })
})
