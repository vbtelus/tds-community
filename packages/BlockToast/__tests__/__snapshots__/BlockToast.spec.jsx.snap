// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockToast renders 1`] = `
<BlockToast__Container
  data-id="block-toast"
  variant="default"
>
  <BlockToast__StyledParagraphWrapper
    variant="default"
  >
    <Text
      block={false}
      bold={false}
      invert={false}
      size="base"
    >
      Save $100 when you shop online
    </Text>
  </BlockToast__StyledParagraphWrapper>
</BlockToast__Container>
`;

exports[`BlockToast should render properly with both copy and link provided 1`] = `
<BlockToast__Container
  data-id="block-toast"
  href="https://www.telus.com"
  text="Learn now"
  variant="default"
>
  <BlockToast__StyledParagraphWrapper
    variant="default"
  >
    <Text
      block={false}
      bold={false}
      invert={false}
      size="base"
    >
      Save $100 when you shop online
    </Text>
  </BlockToast__StyledParagraphWrapper>
</BlockToast__Container>
`;

exports[`BlockToast should render properly with headline, copy and link provided 1`] = `
<BlockToast__Container
  data-id="block-toast"
  variant="default"
>
  <BlockToast__StyledParagraphWrapper
    variant="default"
  >
    <Text
      block={false}
      bold={false}
      invert={false}
      size="base"
    >
      <Strong>
        Limited time offer. 
      </Strong>
      Save $100 when you shop online
      <ChevronLink
        data-id="block-toast-link"
        direction="right"
        href="https://www.telus.com"
        reactRouterLinkComponent={null}
        to={null}
        variant="primary"
      >
         Learn now
      </ChevronLink>
    </Text>
  </BlockToast__StyledParagraphWrapper>
</BlockToast__Container>
`;

exports[`BlockToast should render properly with light variant 1`] = `
<BlockToast__Container
  data-id="block-toast"
  variant="light"
>
  <BlockToast__StyledParagraphWrapper
    variant="light"
  >
    <Text
      block={false}
      bold={false}
      invert={false}
      size="base"
    >
      <Strong>
        Limited time offer. 
      </Strong>
      Save $100 when you shop online
      <ChevronLink
        data-id="block-toast-link"
        direction="right"
        href="https://www.telus.com"
        reactRouterLinkComponent={null}
        to={null}
        variant="primary"
      >
         Learn now
      </ChevronLink>
    </Text>
  </BlockToast__StyledParagraphWrapper>
</BlockToast__Container>
`;

exports[`BlockToast should render properly with toast variant 1`] = `
<BlockToast__Container
  data-id="block-toast"
  variant="toast"
>
  <BlockToast__StyledParagraphWrapper
    variant="toast"
  >
    <Text
      block={false}
      bold={false}
      invert={false}
      size="base"
    >
      <Strong>
        Limited time offer. 
      </Strong>
      Save $100 when you shop online
      <ChevronLink
        data-id="block-toast-link"
        direction="right"
        href="https://www.telus.com"
        reactRouterLinkComponent={null}
        to={null}
        variant="primary"
      >
         Learn now
      </ChevronLink>
    </Text>
  </BlockToast__StyledParagraphWrapper>
</BlockToast__Container>
`;
